model/task.go

```diff
@@ -59,30 +41,37 @@ func NewTaskFromProtobuf(user *User, body *avsproto.CreateTaskReq) (*Task, error
 	aaAddress := *user.SmartAccountAddress
 
 	if body.SmartWalletAddress != "" {
-		if !common.IsHexAddress(body.SmartWalletAddress) {
-			return nil, status.Errorf(codes.InvalidArgument, "invalid smart account address")
-		}
 		aaAddress = common.HexToAddress(body.SmartWalletAddress)
 	}
 
 	taskID := GenerateTaskID()
 
+	if len(body.Edges) == 0 || len(body.Nodes) == 0 {
+		return nil, fmt.Errorf("Missing task data")
+	}
+
 	t := &Task{
 		ID: taskID,
-
-		// convert back to string with EIP55-compliant
-		Owner:              owner.Hex(),
-		SmartWalletAddress: aaAddress.Hex(),
-
-		Trigger:   body.Trigger,
-		Nodes:     body.Actions,
-		Memo:      body.Memo,
-		ExpiredAt: body.ExpiredAt,
-		StartAt:   body.StartAt,
-
-		// initial state for task
-		Status:     avsproto.TaskStatus_Active,
-		Executions: []*avsproto.Execution{},
+		Task: &avsproto.Task{
+			Id: &avsproto.UUID{
+				Bytes: taskID,
+			},
```

Comment:
Isn’t id just a ulid string? I think we should just have `id: string`
instead of return a nested structure `id: { bytes: '01JCKCRTZBS158BRP8E42P2DQW' }`. This nested format doesn’t provide any real benefit to the user and only complicates retrieving the string representation of the ID.

Reply:
Besides, I am using the ulid on the frontend, and find the upper cases are a little hard to read, should we return the id in lower case?

Comparison:
uppercases:  
`{ id: '01JCM1TGR85E59ZYEG5Q7WHKZ6' }`

lowercases:
`{ id: '01jcm1tgr85e59zyeg5q7whkz6' }`

Reply:
okay, let’s keep the upper case for the id, so we don’t care about conversion on the server side. 

yeah, bytes have to be converted to string going through the wires, so I think we should remove the bytes property key, to achieve `{ id: '01JCM1TGR85E59ZYEG5Q7WHKZ6' }`